src/tests/audio/tasks/forced_alignment_test.py

```diff
@@ -0,0 +1,215 @@
+        """Test alignment of segments."""
+        model, processor = dummy_model
+        model_dictionary = processor.tokenizer.get_vocab()
+
+        # Create a sample transcript
+        transcript = [SingleSegment(start=0.0, end=1.0, text="test")]
+
+        # Preprocess the transcript segments
+        preprocessed_transcript = _preprocess_segments(
+            transcript, model_dictionary, model_lang="en", print_progress=False, combined_progress=False
+        )
+
+        # Ensure the model dictionary has the necessary keys
+        for char in "test":
+            if char not in model_dictionary:
+                model_dictionary[char] = len(model_dictionary)
+
+        aligned_segments, word_segments = _align_segments(
+            transcript=preprocessed_transcript,
+            model=model,
+            model_dictionary=model_dictionary,
+            model_lang="en",
+            model_type="huggingface",
+            audio=mono_audio_sample,
+            device=DeviceType.CPU,
+            max_duration=10.0,
+            return_char_alignments=False,
+            interpolate_method="nearest",
+        )
+        assert isinstance(aligned_segments, list)
+        assert isinstance(word_segments, list)
+
+    def test_align_transcription_faked(resampled_mono_audio_sample: Audio, dummy_model: tuple) -> None:
+        """Test alignment of transcription."""
+        model, processor = dummy_model
+        transcript = [
+            SingleSegment(
+                start=0.0,
+                end=1.0,
+                text="test",
+                clean_char=["t", "e", "s", "t"],
+                clean_cdx=[0, 1, 2, 3],
+                clean_wdx=[0],
+                sentence_spans=None,
+            )
+        ]
+        aligned_result = _align_transcription(
+            transcript=transcript,
+            model=model,
+            align_model_metadata={
+                "dictionary": processor.tokenizer.get_vocab(),
+                "language": "en",
+                "type": "huggingface",
+            },
+            audio=resampled_mono_audio_sample,
+            device=DeviceType.CPU,
+        )
+        assert "segments" in aligned_result
+        assert "word_segments" in aligned_result
+
+    def test_align_transcriptions_fixture(resampled_mono_audio_sample: Audio, script_line_fixture: ScriptLine) -> None:
+        """Test alignment of transcriptions."""
+        audios_and_transcriptions = [(resampled_mono_audio_sample, script_line_fixture)]
+        aligned_transcriptions = align_transcriptions(audios_and_transcriptions)
+        assert len(aligned_transcriptions) == 1
+        assert len(aligned_transcriptions[0]) == 1
+        assert aligned_transcriptions[0][0].text == "test"
+
+    def test_align_transcriptions_multilingual(
+        resampled_mono_audio_sample: Audio, script_line_fixture: ScriptLine
+    ) -> None:
+        """Test alignment of transcriptions."""
+        languages = ["de", "es"]
```

Comment:
why is language a string?

Reply:
Now, these are used to generate the Language object which is passed to the function.